packages/in_app_purchase/in_app_purchase_android/migration_guide.md

```diff
@@ -0,0 +1,146 @@
+# Migration Guide from 0.2.x to 0.3.0
+
+Starting November 2023, Android Billing Client V4 is no longer supported,
+requiring app developers using the plugin to migrate to 0.3.0.
+Version 0.3.0 introduces breaking changes to the API as a result of changes in
+the underlying Google Play Billing Library.
+For context around the changes see the [recent changes to subscriptions in Play Console][1]
+and the [Android Billing Client V5 migration guide][2].
+
+## SKUs become Products
+
+SKUs have been replaced with products. For example, the `SkuDetailsWrapper`
+class has been replaced by `ProductDetailsWrapper`, and the `SkuType` enum has
+been replaced by `ProductType`.
+
+Products are used to represent both in-app purchases (also referred to as
+one-time purchases) and subscriptions. There are a few changes to the data model
+that are important to note.
+
+Previously, an SKU of `SkuType.subs` would contain a `freeTrialPeriod`,
+`introductoryPrice` and some related properties. In the new model, this has been
+replaced by a more generic approach.
+A subscription product can have multiple pricing phases, each with its own
+price, subscription period etc. This allows for more flexibility in the pricing
+model.
+
+In the next section we will look at how to migrate code that uses the old model
+to the new model. Instead of listing all possible use cases, we will focus on
+two: getting the price of a one time purchase, and handling free trials for
```

Comment:
nit: Looks like there are three cases here?

Reply:
You're right. Updated.